rbac_api.go

```diff
@@ -276,22 +277,33 @@ func (e *Enforcer) GetImplicitPermissionsForUser(user string, domain ...string)
 // GetImplicitPermissionsForUser("alice") can only get: [["admin", "data1", "read"]], whose policy is default policy "p"
 // But you can specify the named policy "p2" to get: [["admin", "create"]] by    GetNamedImplicitPermissionsForUser("p2","alice")
 func (e *Enforcer) GetNamedImplicitPermissionsForUser(ptype string, user string, domain ...string) ([][]string, error) {
-	roles, err := e.GetImplicitRolesForUser(user, domain...)
-	if err != nil {
-		return nil, err
-	}
-
-	roles = append([]string{user}, roles...)
-
-	var res [][]string
-	var permissions [][]string
-	for _, role := range roles {
-		permissions = e.GetNamedPermissionsForUser(ptype, role, domain...)
-
-		res = append(res, permissions...)
+	permission := make([][]string, 0)
+	rm := e.GetRoleManager()
+	domainIndex := e.getDomainIndex(ptype)
+	for _, rule := range e.model["p"][ptype].Policy {
+		matched := true
+		for _, d := range domain {
+			matched = rm.(*defaultrolemanager.RoleManager).Match(d, rule[domainIndex])
+			if matched {
+				break
+			}
+		}
+		if !matched {
+			continue
+		}
+		matched, _ = rm.HasLink(user, rule[0], domain...)
```

Comment:
``rm.HasLink`` only support one domain. So maybe you should put it in the above loop
```
for _, d := range domain {
	matched = rm.(*defaultrolemanager.RoleManager).Match(d, rule[domainIndex])
	if matched {
		matched, _ = rm.HasLink(user, rule[0], domain...)
	}
}
```

Reply:
done